release/models/system/openconfig-system-grpc.yang

```diff
@@ -97,10 +110,103 @@ module openconfig-system-grpc {
             "Operational state relating to the gRPC service.";
           uses grpc-server-config;
         }
+
+        uses clients-top;
       }
     }
   }
 
+  grouping clients-top {
+    description
+      "Top-level grouping for data related to gNPSI clients.";
+
+    container clients {
+      config false;
+      description
+        "Enclosing container for list of gNPSI clients.";
+
+      list client {
+        key "address port";
+        description
+          "List of gNPSI clients to send sampling data.  Packet
+          samples are sent to all clients specified.";
+
+        leaf address {
+          type leafref {
+            path "../state/address";
+          }
+          description
+            "Reference to address list key.";
+        }
+
+        leaf port {
+          type leafref {
+            path "../state/port";
+          }
+          description
+            "Reference to port list key.";
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for gNPSI clients.";
+
+          uses grpc-server-clients-state;
+
+          container counters {
+            description
+              "Operational data for gNPSI counters.";
+            uses gnpsi-counters-top;
+          }
+        }
+      }
+    }
+  }
+
+  grouping gnpsi-counters-top {
```

Comment:
As I've mentioned previously, suggest following the pattern in https://github.com/openconfig/public/pull/1096 for segmented model design.

gNPSI is a separate optional feature so it should be segmented as such and not included at all within the base gRPC modeling.  There can be many features that have gRPC services along w/ types that can be reused among other models

The intent of the restructuring PR is to provide commonality and reuse where appropriate along w/ well defined demarcations where one can support/not-support model augments vs. needing to produce deviations to a monolith model

Reply:
PR is restructured per reference of  https://github.com/openconfig/public/pull/1096